src/site/content/en/blog/mailru-cwv/index.md

```diff
@@ -0,0 +1,290 @@
+</figure> 
+
+Another effect SSR can have on CLS is the movement components before and after hydration. We encountered this on the mobile version in particular, which led to further layout shifts and negative impacts on CLS, which required special attention to be paid to hydrated component markup. One recurring solution to this problem was to transfer as much display logic from JavaScript to CSS whenever possible.
+
+## Code splitting and unused polyfills
+
+To improve the perceived page load speed, work was required to decrease LCP and FID values. One way to achieve this is through [code splitting](https://web.dev/codelab-code-splitting/). In addition to the Mail.ru home page itself, our team is developing a widget for portal navigation. It is currently embedded in [many of our company's projects](https://vk.company/en/projects/).
+ 
+For historical reasons, the widget is inserted at the very beginning of the page as a synchronously loading script. The share of polyfills in this script grew over time. To limit the negative performance effects of loading these polyfills, we implemented code splitting for both modern and legacy browsers.
+
+We decided against the [`module`/`nomodule` pattern](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/) for loading JavaScript bundles for modern or legacy browsers, as the `<script>` element's [`type="module"` attribute](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Modules) didn't target browsers that were modern enough for our needs. To address this, Mail.ru uses an in-house tool for identifying modern browser versions on the backend, and can adapt to those browsers accordingly.
+ 
+{% Aside %}
+The [Browserslist Useragent](https://github.com/browserslist/browserslist-useragent) package is a publicly available alternative to custom tooling for comparing a project's `.browserslistrc` to the User-Agent string in an application backend. However, our team strongly recommends considering using [User-Agent Client Hints](https://github.com/WICG/ua-client-hints#differential-serving) (for example, [Sec-CH-UA header](https://web.dev/user-agent-client-hints/#user-agent-response-and-request-headers), as Mail.ru does in our in-house tool.
+{% endAside %}
+ 
+Once browsers could be identified in the backend, we could implement code splitting for modern and legacy browsers. This resulted in a 43.3% reduction of the synchronously-loaded JavaScript widget for modern browsers. Also, this practice has been applied to some other portal scripts.
+
+In addition to bundle size reduction and positive effects on CWV, code splitting improves the developer experience, as a 3.5% share of our users use legacy browsers. Since this share of users is on a downward trend, our team decided to rebuild the process of implementation of new features in the widget. This allows developers to use the latest browser APIs despite the bloat that polyfills introduce for users of legacy browsers.
```

Comment:
```suggestion
In addition to bundle size reduction and positive effects on Core Web Vitals, code splitting improves the developer experience, as a 3.5% share of our users use legacy browsers. Since this share of users is on a downward trend, our team decided to rebuild the process of implementation of new features in the widget. This allows developers to use the latest browser APIs despite the bloat that polyfills introduce for users of legacy browsers.
```